src/dmd/dsymbolsem.d

```diff
@@ -6312,3 +6334,122 @@ void aliasSemantic(AliasDeclaration ds, Scope* sc)
             ScopeDsymbol.multiplyDefined(Loc.initial, sx, ds);
     }
 }
+
+/********************************
+ * Semantic of special UDA `gnu_abi_tag`.
+ * Params:
+ *  sc = scope
+ *  s = symbol which has UDAs
+ */
+private void udaGNUAbiTagSemantic(Scope* sc, Dsymbol s)
+{
+    if (!s.userAttribDecl)
+        return;
+    Declaration d = s.isDeclaration();
+    if (d && (d.storage_class & (STC.manifest | STC.templateparameter)))
+        return;
+    auto udas = s.userAttribDecl.getAttributes();
+    expandTuples(udas);
+    auto sc2 = sc.startCTFE();
+    foreach (e; *udas)
+    {
+        assert(e.op != TOK.tuple);
+        if (isUdaGNUAbiTag(e))
+        {
+            {
+                auto lnk = d ? d.linkage : sc.linkage;
+                if (lnk != LINK.cpp)
+                {
+                    s.error("`@%s()` can only apply to C++ symbols", Id.udaGNUAbiTag.toChars());
+                    return;
+                }
+            }
+            if (e.op == TOK.type)
+            {
+                e.error("`@%s` at least one argument expected", Id.udaGNUAbiTag.toChars());
```

Comment:
Won't this be handled automatically?

Reply:
No, that is the case where the UDA is a type i.e. `@type` instead of `@type()`.

Reply:
Hmm, I would expect the compiler to complain if the UDA is a function expecting one argument. But that seems to not be the case.

Reply:
It's useful to have.